libs/sdk-core/src/breez_services.rs

```diff
@@ -574,6 +574,27 @@ impl BreezServices {
         Ok(self.persister.get_payment_by_hash(&hash)?)
     }
 
+    /// Fetch the external metadata of a payment.
+    /// Useful for persisting external, platform-specific information,
+    /// unrelated to the SDK
+    pub async fn payment_external_metadata<T: serde::de::DeserializeOwned>(
+        &self,
+        hash: String,
+    ) -> SdkResult<T> {
+        Ok(self.persister.get_payment_external_metadata(hash)?)
+    }
+
+    /// Set the external metadata of a payment as a valid JSON string
+    pub async fn set_payment_external_metadata(
```

Comment:
NIT: Perhaps we can shorten this to `set_payment_metadata` ?
Also missing the udl file implementation (In case you wanted to hold this until the approach is approved then ignore this comment).